Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt

```diff
@@ -112,10 +117,31 @@ class MainActions(navController: NavHostController) {
     val onboardingComplete: () -> Unit = {
         navController.popBackStack()
     }
-    val selectCourse: (Long) -> Unit = { courseId: Long ->
-        navController.navigate("${MainDestinations.COURSE_DETAIL_ROUTE}/$courseId")
+
+    // Used from COURSES_ROUTE
+    val openCourse = { newCourseId: Long, from: NavBackStackEntry ->
+        // In order to discard duplicated navigation events, we check the Lifecycle
+        if (from.lifecycleIsResumed()) {
+            navController.navigate("${MainDestinations.COURSE_DETAIL_ROUTE}/$newCourseId")
+        }
+    }
+
+    // Used from COURSE_DETAIL_ROUTE
+    val relatedCourse = { newCourseId: Long, from: NavBackStackEntry ->
+        // In order to discard duplicated navigation events, we check the Lifecycle
+        if (from.lifecycleIsResumed()) {
+            navController.navigate("${MainDestinations.COURSE_DETAIL_ROUTE}/$newCourseId")
+        }
     }
-    val upPress: () -> Unit = {
-        navController.navigateUp()
+
+    // Used from COURSE_DETAIL_ROUTE
+    val upPress: (rom: NavBackStackEntry) -> Unit = { from ->
+        // In order to discard duplicated navigation events, we check the Lifecycle
+        if (from.lifecycleIsResumed()) {
+            navController.navigateUp()
+        }
     }
 }
+
+private fun NavBackStackEntry.lifecycleIsResumed() =
```

Comment:
Nit: Can you add a note saying why this is needed when navigating between screens? Is this always needed?